### PR TITLE
Can configure GC in trace replay

### PR DESCRIPTION
**Pull request by Ngoguey42**

### CODE REVIEW

**Comment by Ngoguey42** [comment]
Crashes with #1920 but can be merged

File: bench/irmin-pack/trace_replay_intf.ml [hunk 1]
```diff
@@ -61,7 +64,19 @@ module Config = struct
...
 
       [inode_config] is a pair of ints that will be stored in the results of the
-      replay. *)
+      replay.
+
+      [gc_every] defines the distance between calls to gc. [0] disables the GC.
+
+      [gc_distance_in_the_past] defines the distance between the GC-commit and
+      the latest commit. The first GC will be started after
+      [gc_everygc_distance_in_the_past + 1] commits were replayed. This
```
Comment (tomjridge) [inline bench/irmin-pack/trace_replay_intf.ml:73]: typo? gc_everygc_distance_in_the_past

File: bench/irmin-pack/trace_replay_intf.ml [hunk 2]
```diff
@@ -61,7 +64,19 @@ module Config = struct
       blobs, instead of their actual value read in the trace.
 
       [inode_config] is a pair of ints that will be stored in the results of the
-      replay. *)
+      replay.
+
+      [gc_every] defines the distance between calls to gc. [0] disables the GC.
+
+      [gc_distance_in_the_past] defines the distance between the GC-commit and
```
Comment (tomjridge) [inline bench/irmin-pack/trace_replay_intf.ml:71]: I didn't understand this doc comment for gc_distance_in_the_past initially, but after thinking about it a bit, I think I do. Maybe add something like the following for clarity? (just a suggestion)

GC is trigggered every [gc_every] commits. When GC is triggered, we select a previous commit that is [gc_distance_in_the_past] commits from the current head commit.

**Comment by tomjridge** [comment]
Looks good to me.

**Pull request merged**